tests/cases/fourslash/refactorConvertToOptionalChainExpression_AccessCallCallReturnValue.ts

```diff
@@ -0,0 +1,14 @@
+/// <reference path='fourslash.ts' />
+
+////let a = { b: () => { return () => { c: 0 } } }
+/////*a*/a && a.b && a.b()().c/*b*/;
+
+goTo.select("a", "b");
+edit.applyRefactor({
+    refactorName: "Convert to optional chain expression",
+    actionName: "Convert to optional chain expression",
+    actionDescription: "Convert to optional chain expression",
+    newContent:
+`let a = { b: () => { return () => { c: 0 } } }
+a?.b()()?.c;`

```

Comment:
Given the check `a && a.b`, I don't understand the emit putting the second `?.` right before `c` instead of just after `b`. I expected to see `a?.b?.()().c`

Reply:
Ack. Should it be both? Or would we require `a && a.b && a.b() && a.b().c` as the input (which is currently not allowed due to repeated calls).

Reply:
Consider `a.b && a.b.c.d && a.b.c.d.e.f`. Should we offer a refactor here? If so, should we convert this to `a.b?.c.d?.e.f`, or `a?.b?.c?.d?.e?.f`. If the former, and I think the same goes for the call expression case, we would want to go back to using `isOrContainsMatchingReference` and keep the list of identifiers at the end of each operand so we know which ones to convert.

Reply:
Err, umm, I don't think we should offer a refactor for the `ab, abcd, abcdef` case.
However, `a, ab, ab()()` has a contiguous prefix [1], so it's reasonable to generate `a?.b?.XXX` for it.

Looking at it differently, we should treat `()` the same as properties for the purposes of emit, except that it ends a contiguous prefix. So

```
a && a() --> a?.()
a && a.b && a.b() --> a?.b?.()
// BUT
a && a.b && a.b() && a.b().c --> a?.b?.() && a.b().c // not 100% sure here
```

[1] I am just making up terms and notations here, feel free to propose better ones.

Reply:
I think we had discussed offline that we shouldn't offer a refactor for `a && a.b() && a.b().c` as  `a.b` may not be pure (that may be too conservative?). Is there similar reasoning we can apply to `a && a.b.c`, assuming we convert it to `a?.b.c`?

Reply:
(@jessetrinity explained to me that `a && a.b.c` is valid if `b` doesn't have `undefined` in its type, so it's reasonable to emit `a?.b.c`.)